src/transformState.js

```diff
@@ -5,7 +5,29 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const object of actions) {
+    if (object.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
+
+    if (object.type === 'addProperties') {
+      for (const key in object.extraData) {
+        state[key] = object.extraData[key];
+      }
+    }
```

Comment:
you can use `Object.assign()` and avoid loop here

Reply:
done